### PR TITLE
Fusion Segmenter: Unify single kernel and multi-kernel runtime path

### PR DESCRIPTION
**Pull request by shmsong**
This is a cleanup PR that unifies the two possible runtime paths that a `FusionKernelRuntime` may hold. 

Note that after this PR, the two notions of "is segmented", i.e.:
`segmented_fusion -> isSegmented` and `segmented_fusion -> groups().size() >1` are equivalent. 

There will be two ways a single-kernel fusion will be obtained: 
1. The single kernel fusion heuristic directly accept the in-coming fusion. 
2. The fusion goes through segmenter and still arrive at a single-kernel fusion after translation passes.

TODO: 
 maybe we want to unify these two cases as well for further cleanup. 

Also TODO or maybe in this PR as well,

maybe `SegmentedFusion` -> `FusionGroup` to since it is not necessarily segmented.

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/kernel_cache.cpp [hunk 1]
```diff
@@ -289,40 +290,25 @@ FusionKernelRuntime::FusionKernelRuntime(
     // Take ownership and segment transformed fusion
```
Comment (shmsong) [inline torch/csrc/jit/codegen/cuda/kernel_cache.cpp:290]: Should mostly be code movement and removal of duplicate logic for what used to be the single-kernel path.

File: torch/csrc/jit/codegen/cuda/fusion_segmenter.h [hunk 2]
```diff
@@ -253,6 +253,10 @@ class TORCH_CUDA_CU_API SegmentedFusion {
  public:
   explicit SegmentedFusion(std::unique_ptr<Fusion> fusion);
 
+  static std::unique_ptr<SegmentedFusion> fromCompleteFusion(
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/fusion_segmenter.h:259]: Please add a comment

**Review by naoyam — APPROVED** [review]
Stamping as I don't see any concern

**Review by csarofeen — APPROVED** [review]

File: torch/csrc/jit/codegen/cuda/fusion_segmenter.h (hunk 2 above)
Comment (shmsong) [reply torch/csrc/jit/codegen/cuda/fusion_segmenter.h:259]: Updated. Thanks.

**Pull request merged**